cmake/utils.cmake

```diff
@@ -0,0 +1,369 @@
+# This file is taken from github.com/vllm-project/vllm/cmake/utils.cmake
+# It contains utility functions for building PyTorch extensions with GPU support.
+# The only modification is finding the `Python` package with a specific version.
+#
+# THIS MACRO WAS MODIFIED
+# Attempt to find the python package and verifies that it is one of the `SUPPORTED_VERSIONS`.
+# To customize which python gets found, set the `Python_EXECUTABLE` variable.
+macro (find_python_constrained_versions SUPPORTED_VERSIONS)
```

Comment:
The method of finding the python version in vLLM by using the executable was intentional to avoid ambiguities on systems with multiple python versions.  I think we should use the same logic here to prevent any surprises.

Reply:
The logic has not changed. This just slightly simplifies the logic:
- before, if you had one clear python version or not, you had to manually specify it.
- now, you either have a default that _just works_, or you manually specify if you have to. And `setup.py` always specifies manually

So the only change is for people directly invoking CMake from the terminal that don't have ambiguous Python versions - something I do.

Reply:
I don't see how the logic is the same here.  The old logic intentionally used the exact version derived from `VLLM_PYTHON_EXECUTABLE`.  Afaik `find_package` will just pick the first version of python it can find.

This may not matter much for local development but it matters for CI and packaging,

Reply:
The way I understand this is that `find_package` will find the first Python it can find (which is good in the CMake-from-terminal case) if and only if `Python_EXECUTABLE` is not specified. If `Python_EXECUTABLE` is specified (in the same way `VLLM_PYTHON_EXECUTABLE` was, using `-D`), that version is found. For what it's worth, the old logic manually set `Python_EXECUTABLE` from `VLLM_PYTHON_EXECUTABLE` so this new logic just skips that assignment. And the CI and packaging use `pip install ...` which uses `setup.py` which will always set `Python_EXECUTABLE` anyway. So I believe those cases shouldn't be affected? Am I missing something?

Reply:
I didn't see where `Python_EXECUTABLE` was being set?  setup.py still sets `VLLM_PYTHON_EXECUTABLE`.  Am I missing something?

Reply:
Ohh I see good catch - fixed!